WooCommerce/Classes/ViewRelated/Orders/Order Creation/PaymentSection/FeeOrDiscountLineDetailsViewModel.swift

```diff
@@ -273,3 +283,16 @@ extension FeeOrDiscountLineDetailsViewModel {
         self.amount = priceFieldFormatter.formatUserInput(amountInput)
     }
 }
+
+private extension FeeOrDiscountLineDetailsViewModel {
+    /*
+     The current implementation that we use to calculate a price after discount does not allow to update the discountValueIsDisallowed directly,
+     if we attempt to do so, it would happen while the state/view is being modified, leading to undefined behaviour, and/or crashing.
+     In order to avoid this we defer the state update until the current rendering cycle is complete.
+     */
+    func updateDiscountDisallowedState(_ value: Bool) {
+        DispatchQueue.main.async {
+            self.discountValueIsDisallowed = value
+        }
+    }
```

Comment:
I'm not sure how good practice is this, since we generally send the values directly to the publisher, but I don't see a specific problem with it either. Initially I attempted to use combine to observe changes on the price amount and enable/disable appropriately, but that changes the current implementation enough to open a new can of worms in several places.